hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/service/KeyDeletingService.java

```diff
@@ -235,27 +236,108 @@ private Pair<Pair<Integer, Long>, Boolean> submitPurgeKeysRequest(List<DeleteBlo
         keyToUpdate.addAllKeyInfos(keyInfos);
         keysToUpdateList.add(keyToUpdate.build());
       }
+    }
+
+    int purgeKeyIndex = 0, updateIndex = 0, renameIndex = 0;
+    int currSize = 0;
+    boolean batchCapacityReached;
+
+    while (purgeKeyIndex < purgeKeys.size() ||
+        updateIndex < keysToUpdateList.size() ||
+        (renameEntriesToBeDeleted != null && renameIndex < renameEntriesToBeDeleted.size())) {
+
+      batchCapacityReached = false;
+      PurgeKeysRequest.Builder requestBuilder = PurgeKeysRequest.newBuilder();
 
-      if (!keysToUpdateList.isEmpty()) {
-        purgeKeysRequest.addAllKeysToUpdate(keysToUpdateList);
+      if (snapTableKey != null) {
+        requestBuilder.setSnapshotTableKey(snapTableKey);
+      }
+      if (expectedPreviousSnapshotId != null) {
```

Comment:
This is wrong we are supposed to set the expectedPreviousSnapshotId even if it is null. This could lead to validations not being done on the server side. Please look at the previous code implementation.